build.rs

```diff
@@ -1,4 +1,15 @@
 fn main() {
   println!("cargo:rerun-if-changed=linker.ld");
   println!("cargo:rerun-if-changed=src/asm/entry.S");
+  println!("cargo:rustc-link-arg=-Tlinker.ld");
+
+  // rust's build system sucks. I want to do this, but
+  // the cc, isnt configured the same. It uses a different ABI.
+  //
+  //     cc::Build::new().file("src/asm/entry.S").compile("entry");
```

Comment:
It's fine.

We have `core::arch::global_asm!(include_str!("asm/entry.S"));` right before the `start` function in `main.rs`. It's helpful for visualizing the execution flow :

- We start execution from the `entry` symbol in `entry.S` file.
- Then we enter the `start` function.

Reply:
I know we do, but i want to have a single build file.  Not configs spread around 10 places. The proper place to link an assembly would be build.rs, imo. Inline including a string is just dirty.